src/java.base/share/classes/java/util/concurrent/ForkJoinPool.java

```diff
@@ -1258,12 +1235,12 @@ final boolean tryLockPhase() {    // seqlock acquire
          */
         WorkQueue(ForkJoinWorkerThread owner, int id, int cfg,
                   boolean clearThreadLocals) {
-            if (clearThreadLocals)
-                cfg |= CLEAR_TLS;
-            this.config = cfg;
-            top = base = 1;
-            this.phase = id;
+            array = new ForkJoinTask<?>[owner == null ?
```

Comment:
I like this change—initializing the array itself in the WorkQueue constructor makes a lot of sense.